### PR TITLE
Correctly install service as a LaunchDaemon/LaunchAgent depending on user

### PR DESCRIPTION
**Pull request by nre-ableton**
- Cleanup: remove empty folder
- Assert that Swarm Client user already exists
- Get UID for jenkins_config_owner, not Ansible user
- Rename jenkins_swarm_daemon_plist -> jenkins_swarm_client_launchd_plist
- Set launchd plist file path depending on user
- Version 0.1.2

### CODE REVIEW

**Comment by ala-ableton** [comment]
@nre-ableton I'm interested in reviewing this PR.

File: tasks/darwin/install.yml [hunk 1]
```diff
@@ -15,7 +15,41 @@
   loop_control:
     loop_var: jenkins_swarm_cmd_item
 
-- name: Ensure LaunchDaemon plist file exists
+- name: Set fact for jenkins_swarm_client_legacy_launchd_plist
```
Comment (mst-ableton) [inline tasks/darwin/install.yml:18]: I assume we would remove this after everything is deployed?

**Review by mst-ableton — APPROVED** [review]
lgtm w/ question

File: tasks/darwin/install.yml (hunk 1 above)
Comment (nre-ableton) [reply tasks/darwin/install.yml:18]: Maybe at some point in the future, but it's a public role, so we don't know the migration schedules of others. Sadly, I think we'll probably need to carry these tasks around for awhile.

**Comment by nre-ableton** [comment]
Actually, I need to rethink this change a bit.

**Comment by nre-ableton** [comment]
Bah, so the problem here is that this role can't quite figure out whether it wants to create the Jenkins user or not. :smile: We have tasks to do it, but they're not included from `tasks/main.yml`. When we use this role in our other projects, we don't actually call this file either. Anyways, this is somewhat at odds with af18b0b (_"Assert that Swarm Client user already exists"_).

We should probably make a defaults var to determine if we should create the user, and react accordingly.

File: tasks/main.yml [hunk 2]
```diff
@@ -2,6 +2,10 @@
 - name: Include tasks for sanity checks
   ansible.builtin.import_tasks: "sanity-check.yml"
 
+- name: Create Swarm Client user/group
+  ansible.builtin.import_tasks: "add-jenkins-user.yml"
+  when: swarm_client_create_user
```
Comment (ala-ableton) [inline tasks/main.yml:7]: Line 17 includes `{{ ansible_os_family | lower }}/install.yml`, which includes `unix/add-jenkins-user.yml` or `windows/add-jenkins-user.yml`, so we always include these tasks.

File: tasks/darwin/disable-service.yml [hunk 3]
```diff
@@ -3,7 +3,7 @@
   ansible.builtin.include_vars: "{{ role_path }}/vars/{{ ansible_os_family | lower }}.yml"
 
 - name: Get current UID
```
Comment (ala-ableton) [inline tasks/darwin/disable-service.yml:5]: In 7a0e6b5d4a676a9c471549d3a8db29303c50d688, it might be worth renaming the task, like in https://github.com/Ableton/ansible-role-prometheus-node-exporter/pull/96/commits/d3af7f62123d75e33691210094f12a7280ca437f
(This comment applies to `tasks/darwin/shutdown.yml` and `tasks/darwin/startup.yml` as well)

File: tasks/main.yml (hunk 2 above)
Comment (nre-ableton) [reply tasks/main.yml:7]: Good point. I've reworked this branch a bit and made a separate commit to include this task explicitly, see 31b2c7a.

File: tasks/darwin/disable-service.yml (hunk 3 above)
Comment (nre-ableton) [reply tasks/darwin/disable-service.yml]: Fixed in 779ea65.

File: tasks/unix/add-jenkins-user.yml [hunk 4]
```diff
@@ -1,7 +1,4 @@
 ---
-- name: Include platform-specific defaults
-  ansible.builtin.include_vars: "{{ role_path }}/vars/{{ ansible_os_family | lower }}.yml"
-
```
Comment (ala-ableton) [inline tasks/unix/add-jenkins-user.yml:4]: Removing these lines doesn't seem related to adding the `swarm_client_create_user` default. But these lines also seem to be unused for quite some time... (forever?)

Comment (nre-ableton) [reply tasks/unix/add-jenkins-user.yml:4]: The `include_vars` is necessary if `add-jenkins-user.yml` is called by the playbook via `include_role` (which I think we used to do). But with the introduction of the `swarm_client_create_user` variable, it should never be necessary to call this task file directly, which is why I felt that this change belonged to that commit.

File: defaults/main.yml [hunk 5]
```diff
@@ -41,6 +41,10 @@ jenkins_swarm_home: ""
 # Password for Jenkins service user (only used on Windows).
 jenkins_user_password: ""
 
+# If true, create the user and group for jenkins_config_owner/jenkins_config_group. If
+# false, then this role assumes that this user/group already exist.
```
[the remark this answers is not in the document]
Comment (nre-ableton) [reply defaults/main.yml]: Actually, "exist" would be correct, but "this" should be "the". Anyways, the wording is not great here, so I'll expand. Fixed in 726f59f.

File: tasks/unix/add-jenkins-user.yml (hunk 4 above)
Comment (ala-ableton) [reply tasks/unix/add-jenkins-user.yml:4]: > The `include_vars` is necessary if `add-jenkins-user.yml` is called by the playbook via `include_role`

Why? I don't see what variables from these files are needed.

Comment (nre-ableton) [reply tasks/unix/add-jenkins-user.yml:4]: You're right, it might be for historical reasons, then.

File: tasks/main.yml [hunk 6]
```diff
@@ -2,6 +2,9 @@
 - name: Include tasks for sanity checks
   ansible.builtin.import_tasks: "sanity-check.yml"
 
+- name: Create Swarm Client user/group
+  ansible.builtin.import_tasks: "add-jenkins-user.yml"
```
Comment (ala-ableton) [inline tasks/main.yml:6]: In 787ddc5a24c6cc21df5acb5b48cd689b3116044c, this is done a bit too early. We shouldn't add a Jenkins user if we don't even know the Swarm Client version to install.

Comment (nre-ableton) [reply tasks/main.yml]: I guess it is possible for `get-swarm-client-version.yml` to bail out with an error. Fixed in f2606fe.

**Review by ala-ableton — APPROVED** [review]

**Comment by nre-ableton** [comment]
@ablbot autosquash
